src/main/java/lotto/step2/domain/LottoNumber.java

```diff
@@ -0,0 +1,65 @@
+package lotto.step2.domain;
+
+import java.util.Objects;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+
+    private final static int MIN = 1;
+    private final static int MAX = 45;
+    private final static String MESSAGE_ILLEGAL_NUMBER = "유효범위를 벗어난 숫자입니다.";
+    private final static String MESSAGE_CHECK_NUMERIC = "당첨번호는 정수여야 합니다.";
+    private final static String NUMERIC_REGEX = "^[-]?[0-9]+$";
+
+    private int number;
+
+    public LottoNumber(String number) {
+        this(checkNumeric(number));
+    }
+
+    public LottoNumber(int number) {
+        if (number < MIN || number > MAX) {
+            throw new IllegalArgumentException(MESSAGE_ILLEGAL_NUMBER);
+        }
+        this.number = number;
+    }
+
+    private static int checkNumeric(String number){
+        String numberRegExp = NUMERIC_REGEX;
+        if(!number.matches(numberRegExp)){
+            throw new IllegalArgumentException(MESSAGE_CHECK_NUMERIC);
+        }
+        return Integer.parseInt(number);
+    }
+
+    public int number() {
+        return number;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        LottoNumber otherLottoNumber = (LottoNumber) other;
+        return number == otherLottoNumber.number;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(number);
+    }
+
+    @Override
+    public int compareTo(LottoNumber other) {
+        return number - other.number;
```

Comment:
```suggestion
    public int compareTo(LottoNumber other) {
        return Integer.compare(number, other.number);
```

compareTo 를 Override 할 때는, 기존에 이미 존재하는 compareTo 를 재활용해보는건 어떤가요?
```suggestion
    public int compareTo(LottoNumber other) {
        return number - other.number;
```